src/main/visual_objs.c

```diff
@@ -11,17 +11,74 @@ void func_800AEAC0(struct VisualObj* arg0)
     D_8010A1A0[arg0->base.state](arg0, var_a1);
 }
 
-INCLUDE_ASM("asm/us/main/nonmatchings/visual_objs", func_800AEB1C);
+void func_800AEB1C(struct VisualObj* arg0, struct PlayerObj* arg1)
+{
+    arg0->base.on_screen = 1;
+    arg0->unk38 = 0;
+    arg0->unk3C = (void*)(*(s8**)0x1F80001C + (*(s32**)0x1F80001C)[1]);
+    arg0->unk30 = &D_8011BF40;
+    arg0->unk40 = 0;
+    arg0->unk42 = 0x7804;
+    arg0->base.unk16 = 1;
+    func_800AEC9C(arg0, arg1);
+    func_80015D60(arg0, 3);
+    arg0->base.state++;
+    func_8002B318(arg0, 0x20, 0x20);
+}
 
-INCLUDE_ASM("asm/us/main/nonmatchings/visual_objs", func_800AEBA8);
+void func_800AEBA8(struct VisualObj* arg0, struct PlayerObj* arg1)
+{
+    s32 var_a0 = 0;
 
-INCLUDE_ASM("asm/us/main/nonmatchings/visual_objs", func_800AEC48);
+    func_80015DC8(arg0);
+    if ((*(s32*)&arg1->base.state & 0xFFFF00) == 0xB00) {
```

Comment:
This is an optimization but I can't quite figure it out https://decomp.me/scratch/S8R2M